pyro/ops/matrix_utils.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import, division, print_function
+
+import math
+
+import torch
+from torch.autograd import grad
+
+from pyro.util import warn_if_nan
+
+
+def rinverse(M, sym=False):
```

Comment:
Does this work for all matrices, or only symmetric positive definite matrices?

Reply:
all matrices!